### PR TITLE
feat: enable head component and native typegen

### PR DESCRIPTION
**Pull request by wardpeet**
## Description

Upgrading Gatsby to latest V4:
- adds support to native head component - removes helmet for React 18 compat (https://www.gatsbyjs.com/docs/reference/release-notes/v4.19/#gatsby-head-api)
- adds support to native query typegen (https://www.gatsbyjs.com/docs/reference/release-notes/v4.15/#graphql-typegen)
- adds extra performance boost by upgrading (https://www.gatsbyjs.com/docs/reference/release-notes/v4.14/#improvements-in-image-and-font-loading-times)

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
   "arrowParens": "avoid",
-  "semi": false
-}
+  "semi": false,
```
Comment (wardpeet) [inline .prettierrc:3]: somehow when I ran `yarn format` I got so many changes and this reduced it a little.

File: gatsby-config.js [hunk 2]
```diff
@@ -5,10 +5,9 @@ module.exports = {
       "A query language for your API — GraphQL provides a complete description of the data in your API, gives clients the power to ask for exactly what they need and nothing more, makes it easier to evolve APIs over time, and enables powerful developer tools.",
     siteUrl: "http://graphql.org/",
   },
-
+  graphqlTypegen: true,
```
Comment (wardpeet) [inline gatsby-config.js]: Enables gatsby's typegen

**Comment by Urigo** [comment]
@orta @hwillson i tried to approve the preview in Netlify but couldn't figure out exactly how, do you know what is the latest flow?

**Comment by Urigo** [comment]
I get these options but not sure what they are :)
image

**Comment by hwillson** [comment]
@Urigo I don't have access to `graphql.org`'s Netlify account, but I'm guessing things aren't configured to automatically allow builds from all PR contributors. The further action options are asking how to proceed since they can't validate the PR creator; e.g. selecting the `Approve and add as Git Contributor` option will add @wardpeet to the `graphql.org` Netlify team (details [here](https://docs.netlify.com/accounts-and-billing/team-management/manage-team-members/#git-contributors)). This shouldn't be needed though - there are ways to open this up so all PR's auto-build. Do you know who has admin level access for the `graphql.org` Netlify account? I can help adjust things if we can get access - while we're at it we could also look into getting doc preview builds automatically run as PR checks, with the links auto-added to each PR.

**Comment by Urigo** [comment]
cc @leebyron can you help us change the setting here?

**Comment by Urigo** [comment]
ok I got permissions!
Deployed it and everything looks fine for me
Merging

**Pull request merged**